dallinger/recruiters.py

```diff
@@ -1056,15 +1056,12 @@ def __init__(self, *args, **kwargs):
         self.store = kwargs.get("store") or RedisStore()
         skip_config_validation = kwargs.get("skip_config_validation", False)
 
-        if not skip_config_validation:
-            self._validate_config()
-
-    def _validate_config(self):
-        mode = self.config.get("mode")
-        if mode not in ("sandbox", "live"):
+        if not skip_config_validation and not self.validate_config():
```

Comment:
It is confusing that `validate_config` is being called here as well as in `_deploy_in_mode`. I think it should just be called in the latter.

Reply:
To maintain semantics, we would need to pass `skip_config_validation` to `validate_config`. Why would we want to allow that?

Reply:
If you follow my suggestion of removing `validate_config` from `__init__`, then we can remove `skip_config_validation` entirely from the codebase (I recommend having a look at where it is used).

Reply:
OK, it was only used in methods `hits`, `hit_details`, and `copy_qualifications`. So I removed it now entirely as per your suggestion :+1:

Reply:
Working on the failing test now

Reply:
Presumably you just need to replace

```
MTurkRecruiter()
```
with 

```
MTurkRecruiter().validate_config()
```

or something like that